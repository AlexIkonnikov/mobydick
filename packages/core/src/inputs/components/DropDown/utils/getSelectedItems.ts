import {IListItem} from '../types';

function getSelectedItems<T extends IListItem>(
  selectedItems: T[] | undefined,
  item: T,
) {
  if (selectedItems === undefined || selectedItems.length === 0) {
    return [item];
  }

  const element = selectedItems.find(({value}) => value === item.value);

  if (element) {
    return selectedItems.filter(({value}) => element.value !== value);
  }

  return [...selectedItems, item];
}

export default getSelectedItems;
