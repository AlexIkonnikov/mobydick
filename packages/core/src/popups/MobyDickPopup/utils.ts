import {IPopup} from '../types';

import {IPopupWithProps} from './types';

const isPopupWithProps = (
  it: IPopupWithProps | IPopup,
): it is IPopupWithProps => {
  return Object.prototype.hasOwnProperty.call(it, 'props');
};
export default isPopupWithProps;
