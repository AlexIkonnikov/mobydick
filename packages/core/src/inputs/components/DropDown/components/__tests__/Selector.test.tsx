import {render, fireEvent} from '@testing-library/react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import Selector from '../Selector';
import * as getDropDownDimensions from '../../utils/getDropDownDimensions';

describe('@lad-tech/mobydick-core/Selector', () => {
  const list = [{label: 'list', value: 'list'}];
  const renderItemOnPress = jest.fn();
  const largeList = [
    {label: '1', value: '1'},
    {label: '1', value: '1'},
    {label: '1', value: '1'},
    {label: '1', value: '1'},
    {label: '1', value: '1'},
    {label: '1', value: '1'},
    {label: '1', value: '1'},
  ];

  const selectedItem = list[0]?.value;

  it('renders correctly', async () => {
    const inputList = [{label: 'list', value: 'list'}] as const;
    const {toJSON, findByLabelText} = render(
      <SafeAreaProvider>
        <Selector
          list={[...inputList]}
          selectedItem={list[1]?.value}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          dropDownHeight={80}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();

    const item = await findByLabelText(inputList[0].label);
    fireEvent.press(item);

    expect(renderItemOnPress).toHaveBeenCalledWith(inputList[0]);
  });
  it('renders correctly with bottom padding largeList', async () => {
    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={largeList}
          onClose={jest.fn()}
          pageY={1}
          selectedItem={largeList[1]?.value}
          renderItemOnPress={renderItemOnPress}
          dropDownHeight={80}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('render correct renderItem TouchableHighlight with custom style', () => {
    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={list}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          flatListItemStyle={{height: 10}}
          selectedItem={selectedItem}
          selectedItemColor={'red'}
          dropDownHeight={80}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('render correct renderItem Typography with custom style', () => {
    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={list}
          selectedItem={selectedItem}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          flatListTextStylePressed={{backgroundColor: 'red'}}
          dropDownHeight={80}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('render correct renderItem Typography  with custom font', () => {
    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={list}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          selectedItem={selectedItem}
          flatListTextFontPressed={'Bold-Error-L'}
          dropDownHeight={80}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('How props Selector affect getDropDownDimensions', () => {
    const spy = jest.spyOn(getDropDownDimensions, 'getDropDownDimensions');
    render(
      <SafeAreaProvider>
        <Selector
          list={list}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          selectedItem={selectedItem}
          flatListStyle={{paddingVertical: 2}}
          dropDownHeight={72}
        />
      </SafeAreaProvider>,
    );

    expect(spy).toHaveBeenLastCalledWith({
      dropDownBorderWidth: 1.5,
      dropDownHeight: 72,
      height: 640,
      navBarHeight: 50,
      pageY: 1,
      listLength: 1,
    });
  });

  it('Selector custom flatList styles', () => {
    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={list}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          selectedItem={selectedItem}
          flatListStyle={{width: 2}}
          buttonStyle={{borderWidth: 1}}
          dropDownHeight={72}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('Selector custom with array in selectedItem', () => {
    const data = [
      {label: 'JavaScript', value: 1},
      {label: 'Rust', value: 2},
    ];

    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={data}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          selectedItem={[data[0]!]}
          dropDownHeight={72}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('Selector custom with empty array in selectedItem', () => {
    const data = [
      {label: 'JavaScript', value: 1},
      {label: 'Rust', value: 2},
    ];

    const {toJSON} = render(
      <SafeAreaProvider>
        <Selector
          list={data}
          onClose={jest.fn()}
          pageY={1}
          renderItemOnPress={renderItemOnPress}
          selectedItem={[]}
          dropDownHeight={72}
        />
      </SafeAreaProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
