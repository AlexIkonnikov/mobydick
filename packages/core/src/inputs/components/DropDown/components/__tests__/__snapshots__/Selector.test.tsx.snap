// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/Selector Selector custom flatList styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        {
          "width": 2,
        },
        {
          "width": 2,
        },
        {
          "top": 79,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "list",
            "value": "list",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="list"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              list
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/Selector render correct renderItem TouchableHighlight with custom style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        undefined,
        {
          "width": 300,
        },
        {
          "top": 87,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "list",
            "value": "list",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="list"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                {
                  "height": 10,
                },
                {
                  "backgroundColor": "red",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              list
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/Selector render correct renderItem Typography  with custom font 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        undefined,
        {
          "width": 300,
        },
        {
          "top": 87,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "list",
            "value": "list",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="list"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#F54D3D",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 27,
                    "lineHeight": 33,
                    "minHeight": 33,
                  },
                  undefined,
                ]
              }
            >
              list
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/Selector render correct renderItem Typography with custom style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        undefined,
        {
          "width": 300,
        },
        {
          "top": 87,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "list",
            "value": "list",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="list"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  {
                    "backgroundColor": "red",
                  },
                ]
              }
            >
              list
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/Selector renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        undefined,
        {
          "width": 300,
        },
        {
          "top": 87,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "list",
            "value": "list",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="list"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                null,
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#444B5A",
                    "fontFamily": "Inter-Regular",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              list
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/Selector renders correctly with bottom padding largeList 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="Pressable"
>
  <View
    style={
      [
        {
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.15)",
          "borderRadius": 18,
          "borderWidth": 1.5,
          "paddingVertical": 12,
          "position": "absolute",
        },
        undefined,
        {
          "width": 300,
        },
        {
          "top": 87,
        },
        {
          "maxHeight": 366,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptySelector />}
      bounces={false}
      data={
        [
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
          {
            "label": "1",
            "value": "1",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityLabel="1"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "justifyContent": "center",
                  "paddingHorizontal": 18,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "backgroundColor": "rgba(43, 120, 238, 0.15)",
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                [
                  {
                    "color": "#20242D",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 24,
                    "lineHeight": 30,
                    "minHeight": 30,
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
